core/domain/story_domain.py

```diff
@@ -95,7 +95,8 @@ def __init__(self, change_dict):
         self.cmd = change_dict['cmd']
 
         if self.cmd == CMD_ADD_STORY_NODE:
-            self.node_id = change_dict['node_id']
+            # This command doesn't need any parameters.
```

Comment:
This is interesting. I just realized something -- if we want our changelists to be easily used for reverting, we might need to specify the node ID, otherwise it's hard to tell which node has been added.

So, what do you think of this approach: send the next_node_id for a story to the story editor frontend as well, and let the frontend generate the appropriate commands, with node IDs, when adding story nodes (in fact, thinking about it, the frontend will probably need to have the node ID anyway). Then when sending these to the backend and adding the node, what add_node() should do is to take in a desired_node_id arg and compare it with next_node_id, raise an error if the two differ, and apply the change otherwise.

What do you think?

Reply:
Done, also added tests for the same.